lambda/shared/src/main/scala/feral/lambda/TracedLambda.scala

```diff
@@ -40,4 +43,18 @@ object TracedLambda {
     }
   } yield result
 
+  def apply[Event, Result](entryPoint: EntryPoint[IO])(lambda: Trace[IO] => IO[Option[Result]])(
+      implicit env: LambdaEnv[IO, Event],
+      KS: KernelSource[Event]): IO[Option[Result]] = for {
```

Comment:
I'm a bit concerned this will be ambiguous with the other constructor. Let me add some tests.